packages/livekit-rtc/src/audio_source.ts

```diff
@@ -47,7 +63,50 @@ export class AudioSource {
     this.ffiHandle = new FfiHandle(res.source.handle.id);
   }
 
+  get queuedDuration(): number {
+    return Math.max(
+      this.currentQueueSize - Number(process.hrtime.bigint() / 1000000n) + this.lastCapture,
+      0,
+    );
+  }
+
+  clearQueue() {
+    const req = new ClearAudioBufferRequest({
+      sourceHandle: this.ffiHandle.handle,
+    });
+
+    FfiClient.instance.request<ClearAudioBufferResponse>({
+      message: {
+        case: 'clearAudioBuffer',
+        value: req,
+      },
+    });
+
+    this.releaseQueue.put();
```

Comment:
if clearQueue is called multiple times, it seems like waitForPlayout will directly returns?

Reply:
right this isn't one-to-one. i'll rework this to use Futures